### PR TITLE
Extract JdbcDiagnosticModule

### PR DESCRIPTION
**Pull request by kokosing**
Extract JdbcDiagnosticModule

### CODE REVIEW

**Comment by kokosing** [comment]
Use `JdbcDiagnosticModule` in `PhoenixModule`

File: presto-base-jdbc/src/main/java/io/prestosql/plugin/jdbc/JdbcDiagnosticModule.java [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+import io.prestosql.plugin.jdbc.jmx.StatisticsAwareConnectionFactory;
+import io.prestosql.plugin.jdbc.jmx.StatisticsAwareJdbcClient;
+import org.weakref.jmx.guice.MBeanModule;
+
+import static com.google.common.reflect.Reflection.newProxy;
+import static java.lang.String.format;
+import static java.util.Objects.requireNonNull;
+import static org.weakref.jmx.guice.ExportBinder.newExporter;
+
+public class JdbcDiagnosticModule
```
Comment (kokosing) [inline presto-base-jdbc/src/main/java/io/prestosql/plugin/jdbc/JdbcDiagnosticModule.java]: Use `JdbcDiagnosticModule` in `PhoenixModule`

File: presto-base-jdbc/src/main/java/io/prestosql/plugin/jdbc/JdbcDiagnosticModule.java [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+
+        newExporter(binder).export(Key.get(JdbcClient.class, StatsCollecting.class))
+                .as(generator -> generator.generatedNameOf(JdbcClient.class, catalogName));
+        newExporter(binder).export(Key.get(ConnectionFactory.class, StatsCollecting.class))
+                .as(generator -> generator.generatedNameOf(ConnectionFactory.class, catalogName));
+    }
+
+    @Provides
+    @Singleton
+    @StatsCollecting
```
Comment (sopel39) [inline presto-base-jdbc/src/main/java/io/prestosql/plugin/jdbc/JdbcDiagnosticModule.java]: who does final, unannotated binding now?

Comment (kokosing) [reply presto-base-jdbc/src/main/java/io/prestosql/plugin/jdbc/JdbcDiagnosticModule.java]: In `JdbcModule`:
```
        binder.bind(JdbcClient.class)
                .to(Key.get(JdbcClient.class, StatsCollecting.class));
        binder.bind(ConnectionFactory.class)
                .to(Key.get(ConnectionFactory.class, StatsCollecting.class));
```

**Pull request merged**